ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutor.scala

```diff
@@ -61,70 +63,71 @@ private[apiserver] final class LedgerTimeAwareCommandExecutor(
           val usedContractIds: Set[ContractId] = cer.transaction
             .inputContracts[ContractId]
             .collect { case id: ContractId => id }
-          if (usedContractIds.isEmpty)
-            Future.successful(Right(cer))
-          else
-            contractStore
-              .lookupMaximumLedgerTime(usedContractIds)
-              .flatMap { maxUsedTime =>
-                if (maxUsedTime.forall(_ <= commands.commands.ledgerEffectiveTime)) {
-                  Future.successful(Right(cer))
-                } else if (!cer.dependsOnLedgerTime) {
-                  logger.debug(
-                    s"Advancing ledger effective time for the output from ${commands.commands.ledgerEffectiveTime} to $maxUsedTime"
-                  )
-                  Future.successful(Right(advanceOutputTime(cer, maxUsedTime)))
-                } else if (retriesLeft > 0) {
-                  metrics.daml.execution.retry.mark()
+
+          def failed = Future.successful(Left(ErrorCause.LedgerTime(maxRetries - retriesLeft)))
+          def success(c: CommandExecutionResult) = Future.successful(Right(c))
+          def retry(c: Commands) = {
+            metrics.daml.execution.retry.mark()
+            loop(c, submissionSeed, ledgerConfiguration, retriesLeft - 1)
+          }
+
+          contractStore
+            .lookupMaximumLedgerTimeAfterInterpretation(usedContractIds)
+            .transformWith {
+              case Success(MaximumLedgerTime.NotAvailable) =>
+                success(cer)
+
+              case Success(MaximumLedgerTime.Max(maxUsedTime))
+                  if maxUsedTime <= commands.commands.ledgerEffectiveTime =>
+                success(cer)
+
+              case Success(MaximumLedgerTime.Max(maxUsedTime)) if !cer.dependsOnLedgerTime =>
+                logger.debug(
+                  s"Advancing ledger effective time for the output from ${commands.commands.ledgerEffectiveTime} to $maxUsedTime"
+                )
+                success(advanceOutputTime(cer, maxUsedTime))
+
+              case Success(MaximumLedgerTime.Max(maxUsedTime)) =>
+                if (retriesLeft > 0) {
                   logger.debug(
                     s"Restarting the computation with new ledger effective time $maxUsedTime"
                   )
-                  val advancedCommands = advanceInputTime(commands, maxUsedTime)
-                  loop(advancedCommands, submissionSeed, ledgerConfiguration, retriesLeft - 1)
+                  retry(advanceInputTime(commands, maxUsedTime))
                 } else {
-                  Future.successful(Left(ErrorCause.LedgerTime(maxRetries)))
+                  failed
```

Comment:
The omission to log anything in this case is intriguing

Reply:
let me try to indulge you ;) that will boil down to a plain too many retries error, which is exposed as errorcodes with proper logging already